tests-e2e/core/gitopsdeployment_syncrun_test.go

```diff
@@ -470,6 +471,55 @@ var _ = Describe("GitOpsDeploymentSyncRun E2E tests", func() {
 				Expect(app.Status.History[i].Revision).Should(Equal(cr.revision))
 			}
 		})
+
+		It("should refresh before syncing the Application", func() {
```

Comment:
To verify whether Argo CD refreshes the app, I've added the refresh annotation here. Although there is no control over when Argo CD removes the annotation and this might be a race condition, we can still verify if Argo CD behaves as expected. Let me know if there is a better way to test this feature.

Reply:
Yup, agreed, it's very tough to test as part of an E2E test.